chsdi/__init__.py

```diff
@@ -49,49 +73,276 @@ def main(global_config, **settings):
-    config.add_route('legend', '/rest/services/{map}/MapServer/{layerId}/legend')
-    config.add_route('releases', '/rest/services/{map}/MapServer/{layerId}/releases')
-    config.add_route('cacheUpdate', '/rest/services/{map}/MapServer/{layerId}/cacheUpdate')
-    config.add_route('featureAttributes', '/rest/services/{map}/MapServer/{layerId}')
-    config.add_route('feature', '/rest/services/{map}/MapServer/{layerId}/{featureId}')
-    config.add_route('htmlPopup', '/rest/services/{map}/MapServer/{layerId}/{featureId}/htmlPopup')
-    config.add_route('iframeHtmlPopup', '/rest/services/{map}/MapServer/{layerId}/{featureId}/iframeHtmlPopup')
-    config.add_route('extendedHtmlPopup', '/rest/services/{map}/MapServer/{layerId}/{featureId}/extendedHtmlPopup')
-    config.add_route('search', '/rest/services/{map}/SearchServer')
-    config.add_route('wmtscapabilities', '/rest/services/{map}/1.0.0/WMTSCapabilities.xml')
-    config.add_route('feedback', '/feedback')
-    config.add_route('qrcodegenerator', '/qrcodegenerator')
-    config.add_route('sitemap', '/sitemap')
-    config.add_route('luftbilder', '/luftbilder/viewer.html')
-    config.add_route('historicalmaps', '/historicalmaps/viewer.html')
-    config.add_route('checker', '/checker')
-    config.add_route('checker_dev', '/checker_dev')
-    config.add_route('downloadkml', '/downloadkml')
-    config.add_route('files_collection', '/files')
-    config.add_route('files', '/files/{id}')
-    config.add_route('adminkml', '/admin/kml')
-    config.add_route('stationboard', '/stationboard/stops/{id}')
-    config.add_route('faqlist', '/rest/services/{map}/faqlist')
-    config.add_route('cut', '/rest/services/{map}/GeometryServer/cut')
-    config.add_route('color', '/color/{r},{g},{b}/{image}')
-
-    # Some views for specific routes
-    config.add_view(route_name='dev', renderer='chsdi:templates/index.mako')
-    config.add_view(route_name='testi18n', renderer='chsdi:templates/testi18n.mako')
+    # Dev page with bunch of demo links
+    config.add_route('dev', '/dev', request_method='GET')
+    # JS API loader
+    config.add_route('ga_api', '/loader.js', request_method='GET')
+    add_cors_route(config, '/loader.js', 'ga_api', methods=['GET'])
 
+    config.add_route('testi18n', '/testi18n', request_method='GET')
+    # List topics
+    config.add_route('topics', '/rest/services', request_method='GET')
+    add_cors_route(config, '/rest/services', 'topics', methods=['GET'])
+    # Returns layers metdata
+    config.add_route('mapservice', '/rest/services/{map}/MapServer',
+                     request_method='GET')
+    add_cors_route(config,
+                   '/rest/services/{map}/MapServer',
+                   'mapservice',
+                   methods=['GET'])
+    # Returns the technical configuration of the layers in geoadmin
+    config.add_route('layersConfig',
+                     '/rest/services/{map}/MapServer/layersConfig',
+                     request_method='GET')
```

Comment:
Sometimes `['GET']`, sometimes `'GET'`.

Reply:
it's the same, It'll fix to be consistent.

Reply:
actually `request_method` is always a string, except when we have several methods.

`methods` defines a custom route for `HEAD` and `OPTIONS` requests.

this is the way to do it in pyramid. As much as it is ugly....
c2c geoportal does something similar.